crypto/txscript/src/opcodes/mod.rs

```diff
@@ -883,12 +883,82 @@ opcode_list! {
             _ => Err(TxScriptError::InvalidSource("LockTimeVerify only applies to transaction inputs".to_string()))
         }
     }
+    opcode OpInputSpk<0xb2, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{
+                    utxo_entry: kaspa_consensus_core::tx::UtxoEntry{
+                        script_public_key: spk @ kaspa_consensus_core::tx::ScriptPublicKey{
+                            version, ..},
+                        ..
+                    },
+                    ..
+                } => {
+                    let version = version.to_be_bytes();
+                    let script = spk.script();
+                    let mut v = Vec::with_capacity(version.len() + script.len());
+                    v.extend_from_slice(&version);
+                    v.extend_from_slice(script);
+                    vm.dstack.push(v);
+                    Ok(())
+                },
+                _ => Err(TxScriptError::InvalidSource("OpInputSpk only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpInputAmount<0xb3, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{
+                    utxo_entry: kaspa_consensus_core::tx::UtxoEntry{
+                        amount,
+                        ..
+                    },
+                    ..
+                } => push_number(*amount as i64, vm),
+                _ => Err(TxScriptError::InvalidSource("OpInputAmount only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpOutputSpk<0xb4, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{tx, id , ..} => {
+                    let v = tx.outputs().get(id).map(|output| {
+                        let version =  output.script_public_key.version.to_be_bytes();
+                        let script = output.script_public_key.script();
+                        let mut v = Vec::with_capacity(version.len() + script.len());
+                        v.extend_from_slice(&version);
+                        v.extend_from_slice(script);
+                        v
+                    });
+                    vm.dstack.push(v.unwrap_or_default());
```

Comment:
It should be documented in the KIP that empty slice is pushed if idx is out of bound

Reply:
Check the latest changes. The error is there. Input len, output Len and current input index will be added